src/Agent.Worker/ExecutionContext.cs

```diff
@@ -444,10 +446,11 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
             {
                 DefaultStepTarget = HostContext.CreateContainerInfo(message.Resources.Containers.Single(x => string.Equals(x.Alias, message.JobContainer, StringComparison.OrdinalIgnoreCase)));
                 Containers.Add(DefaultStepTarget);
-                foreach (var container in message.Resources.Containers.FindAll(x => !string.Equals(x.Alias, message.JobContainer, StringComparison.OrdinalIgnoreCase)))
-                {
-                    Containers.Add(HostContext.CreateContainerInfo(container));
-                }
+            }
+            // Include other step containers
+            foreach (var container in message.Resources.Containers.FindAll(x => !string.Equals(x.Alias, message.JobContainer, StringComparison.OrdinalIgnoreCase)))
```

Comment:
"Where" might be a better choice here instead of "FindAll". FindAll is basically a Where combined with a ToList. Since you aren't keeping the list, a where is a bit better since it just returns an enerable.